pipelined/bevy_render2/src/shader/shader_reflect.rs

```diff
@@ -395,20 +436,35 @@ mod tests {
                                 property: UniformProperty::Struct(vec![UniformProperty::Mat4]),
                             },
                             shader_stage: BindingShaderStage::VERTEX,
+                            count: None,
                         }]
                     ),
                     BindGroupDescriptor::new(
                         1,
-                        vec![BindingDescriptor {
-                            index: 0,
-                            name: "Texture".into(),
-                            bind_type: BindType::Texture {
-                                multisampled: false,
-                                view_dimension: TextureViewDimension::D2,
-                                sample_type: TextureSampleType::Float { filterable: true }
+                        vec![
+                            BindingDescriptor {
+                                index: 0,
+                                name: "Texture".into(),
+                                bind_type: BindType::Texture {
+                                    multisampled: false,
+                                    view_dimension: TextureViewDimension::D2,
+                                    sample_type: TextureSampleType::Float { filterable: true }
+                                },
+                                shader_stage: BindingShaderStage::VERTEX,
+                                count: None,
                             },
-                            shader_stage: BindingShaderStage::VERTEX,
-                        }]
+                            BindingDescriptor {
+                                index: 1,
+                                name: "TextureArr".into(),
+                                bind_type: BindType::Texture {
+                                    multisampled: false,
+                                    view_dimension: TextureViewDimension::D2Array,
```

Comment:
I thought D2Array was for array textures not array of textures. That is, texture2DArray in a glsl shader.

Reply:
I think you are right. Weirdly I didn't get any vulkan validation errors with D2Array, but neither do I with D2. Updated.